### ISSUE TITLE
How can we drag and clone an element stand outside into the grid?

### ISSUE THREAD

**Issue by wanghan-js** [haltu/muuri#442]
I really love muuri. It's beautiful and elegant.

I read some source code of it, and found it concise and clear.

The README.md says that muuri can do any layout I want.

But I find there is one feature I could not have it done: how can we drag and clone an element stand outside into the grid?

Please see the demo of gridstack: https://gridstackjs.com/demo/advance.html

What I want is drag the green square on the left side which says 'Drag me in the dashboard!' in the right dashboard.

While I'm dragging it, it can participate in the layout of the dashboard, just like I'm drag and drop an item which is already in the dashboard.

It's not like dragging an item from a grid and drop it in another grid. Basically, when I drag an ordinary dom element (dragstart), it spawns an element of whatever size, and which moves along with the pointer, participates in the layout of the grid, at last drops in the grid at a reasonable position.

Could you please tell me how to achieve the effect above? I've read the docs of muuri but didn't find an answer.

Many thanks!

**Comment by niklasramo** [comment]
@wanghan-js that's actually something that Muuri does not handle out of the box, but thinking about it I guess you could hack it together (not so elegantly) with custom layout and dragSortPredicate. When you control the layout fully yourself you can make room for the external dragged element when it's hovering over the grid and when you drop the element you can then add it to the grid making it a proper grid item.

**Comment by wanghan-js** [comment]
@niklasramo Thanks for your prompt reply. But I still don't understand how to do that exactly. I mean, when I dragged an element in the grid, maybe I can add an new item and possibly make it move in the grid, but I can't really **drag** that new item, what I am dragging is always the outside element, so the grid can't make a placeholder for me, and the new item will not emit the drag events.

Could you please make a simple CodePen example to illustrate your approach? Please accept my deepest thanks for this issue.

**Comment by niklasramo** [comment]
Yep, the dragged external element will not trigger any events and will not have any other benefits of a grid item, but as I mentioned it's a not so elegant hack. Maybe you could add it as a grid item when the element enters the grid though, might work? Unfortunately due to time constraints I can't make a demo for this now, maybe some other time in the future.
